Lib/pprint.py

```diff
@@ -589,6 +589,37 @@ def _safe_repr(self, object, context, maxlevels, level):
             del context[objid]
             return format % ", ".join(components), readable, recursive
 
+        if (issubclass(typ, set) and r is set.__repr__) or \
```

Comment:
Sorry, but this block is almost a complete duplicate of the one above (please fix that).

Reply:
I have merged this section with the above one. Please let me know if this is satisfactory. I originally separated them to follow the layout of the function and separate out new code from old code.